src/data/compressedlinestorage.cpp

```diff
@@ -85,7 +80,7 @@ namespace {
         // Stored in big endian format in order to recognise the initial pattern:
         // 10xx xxxx xxxx xxxx
         //  HO byte | LO byte
-        *(reinterpret_cast<uint16_t*>(*ptr)) = htons( value | (1 << 15) );
+        *(reinterpret_cast<uint16_t*>(*ptr)) = qToBigEndian( static_cast<uint16_t>( value | (1 << 15) ) );
```

Comment:
Without static_cast some ugly conversions break everything on first not short line )